h/services/bulk_annotation.py

```diff
@@ -0,0 +1,100 @@
+from typing import List
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+from sqlalchemy.sql import Select
+
+from h.db.query_filters import date_match
+from h.models import Annotation, AnnotationModeration, Group, User
+
+
+class BulkAnnotationService:
+    """A service for retrieving annotations for many users at once."""
+
+    def __init__(self, db_session: Session):
+        """Initialise the service."""
+
+        self._db = db_session
+
+    def annotation_search(
+        self, authority: str, audience: dict, updated: dict, limit=100000
+    ) -> List[Annotation]:
+        """
+        Get a list of annotations viewable by a particular set of users
+
+        :param authority: The authority to search by
+        :param audience: A specification of how to find the users (e.g)
+            {"username": [...]}
```

Comment:
I think the dict-containing-a-list-of-usernames format might be an API thing rather than a business logic thing, so possibly `audience` here should just be a list of user objects or username strings or whatever is needed, and it should be the view's job to unpack that from the JSON request body.